### PR TITLE
Desktop release v0.3.4

### LINKED ISSUE #260: [MacOS] App silently fails after denied keychain access
### Bug description
After Denying access to the keychain the application wont accept passwords for unlocking or saving them. It doesn't throw an error, just fails silently and appears unresponsive. 

### Environment Specs
What device are you running Trinity on?

- Manufacturer: MacOS
- Model: SuperDooper
- OS version: High Sierra

What build of Trinity are you running? Please provide the full version (you can find it in Settings > About Trinity).

- Trinity version: 0.3.1 Desktop

### Steps To Reproduce
1. Remove Trinity keychain entries
2. When prompted, deny access
3. Try to enter a password

### Expected behaviour
An error throws to the user when trying to access a keychain that it doesn't have permissions for.

### Actual behaviour
Nada

### Errors
Error

### Description
A toast notification for the error and a link? to how to fix it

### Motivation
To stop users whining when the app is inexplicably in between them and their tokens

### Open Questions (optional)
Why do people say "tuna fish" when they don't say "beef mammal" or "chicken bird"?

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

- New: Drag&amp;drop text seed support (https://github.com/iotaledger/trinity-wallet/pull/396)
- Update: Trigger 2fa verification once necessary code string length is reached (https://github.com/iotaledger/trinity-wallet/pull/353)
- Update: Seed in memory use update (https://github.com/iotaledger/trinity-wallet/pull/350)
- Update: Adjusted auto promotion timing (https://github.com/iotaledger/trinity-wallet/pull/368)
- Fix: Reattach only if transaction falls below max depth (https://github.com/iotaledger/trinity-wallet/pull/371)
- Fix: Filled and empty paper wallet templates are mixed (https://github.com/iotaledger/trinity-wallet/pull/349)
- Fix: On Windows resizing wallet window using top corners does not work (https://github.com/iotaledger/trinity-wallet/pull/352)

Fixes #260 #318 #308 #320 #324 

# How Has This Been Tested?

Build tested on macOS and Ubuntu

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

**Review by cvarley100 — APPROVED** [review]

File: src/desktop/package.json [hunk 1]
```diff
@@ -136,24 +136,25 @@
     "curl.lib.js": "^1.0.22",
     "date-fns": "^1.29.0",
     "electron-settings": "^3.2.0",
-    "electron-updater": "^3.0.3",
+    "electron-updater": "^3.1.2",
     "entangled-node": "rihardsgravis/entangled-node",
-    "i18next": "^11.5.0",
+    "i18next": "^11.9.0",
+    "iota.lib.js": "^0.5.1",
```
Comment (rajivshah3) [inline src/desktop/package.json:142]: Isn’t 0.5.0 the latest version?

**Comment by laumair** [comment]
@rihardsgravis `iota.lib.js` in `shared/` still has `0.5.0`. Do we want to update that to `0.5.1`?

File: src/desktop/package.json (hunk 1 above)
Comment (laumair) [reply src/desktop/package.json:142]: `0.5.1` is the latest

Comment (rihardsgravis) [reply src/desktop/package.json:142]: According to https://github.com/iotaledger/iota.js/blob/develop/package.json it's `0.5.1`

Comment (rajivshah3) [reply src/desktop/package.json:142]: Whoops, didn’t realize there was a new version on npm. I don’t think there was a tag for this one though

**Review by rajivshah3 — APPROVED** [review]

**Pull request merged**